src/components/Fields/DateField.js

```diff
@@ -0,0 +1,165 @@
+import React, { Component } from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import StyleObjectPropType from '../../prop-types/style';
+import TextField from './TextField';
+import DateInlinePicker from './DateInlinePicker';
+import Tooltip from '../Tooltip/Tooltip';
+import { TooltipPosition } from '../Tooltip/TooltipEnums';
+import { DateFieldRangePosition } from './TextFieldEnums';
+import './DateField.scss';
+
+const propTypes = {
+    className: PropTypes.string,
+    forceHideCalendar: PropTypes.bool,
+    isRange: PropTypes.bool,
+    maxDate: PropTypes.instanceOf(Date),
+    minDate: PropTypes.instanceOf(Date),
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func,
+    onFocus: PropTypes.func,
+    rangePosition: PropTypes.oneOf([
+        DateFieldRangePosition.START,
+        DateFieldRangePosition.FINISH,
+    ]),
+    style: StyleObjectPropType(),
+    value: PropTypes.oneOfType([
+        PropTypes.instanceOf(Date),
+        PropTypes.arrayOf(PropTypes.instanceOf(Date)),
+    ]),
+};
+
+const defaultProps = {
+    className: '',
+    isRange: false,
+    forceHideCalendar: false,
+    maxDate: null,
+    minDate: null,
+    onBlur: null,
+    onChange: null,
+    onFocus: null,
+    rangePosition: DateFieldRangePosition.START,
+    style: null,
+    value: null,
+};
+
+class DateField extends Component {
+    state = {
+        selectedDate: this.props.value,
+        showDatePicker: false,
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.value !== nextProps.value) {
+            this.setState({ selectedDate: nextProps.value });
+        }
+        if (this.props.forceHideCalendar !== nextProps.forceHideCalendar) {
+            if (nextProps.forceHideCalendar) {
+                this.setState({ showDatePicker: false });
+            }
+        }
+    }
+
+    clearInputAfterChange = (inputValue) => {
+        if (inputValue === null || inputValue === '') {
+            this.setState({ selectedDate: null });
+        }
+    }
+
+    datePickerClose() {
+        if (this.props.isRange === false ||
+            (this.state.selectedDate && this.state.selectedDate.length === 2)
+        ) {
+            this.setState({
+                showDatePicker: false,
+            });
+        }
+    }
+
+    datePickerOpen() {
+        this.setState({
+            showDatePicker: true,
+        });
+    }
+
+    formatDate = (date) => {
+        if (!date) {
+            return date;
+        }
+        const dateGroupsRegex = /\w{3} (\w{3}) (\d{2}) (\d{4}) .+/;
+        return date.toString().replace(dateGroupsRegex, '$2 $1 $3');
```

Comment:
Can we use a date library for formatting so it's more explicit what you're trying to do?

Reply:
Discussed a date library on the #devs channel... Jim and others are VERY keen to avoid a date library.

I picked Flatpickr (the underlying calendar) because it uses the Date object and nothing more – I wanted to keep the component light and allow the features lanes to pick their own date library (if they want one).

My theory was that we may already have a date library in ST* and I didn't want to go and add another.

With that said I agree that bit of regex is fairly minging... options:

1. add a date library
2. add a comment to explain
3. leave it as is

What would you like to do?

* I checked ST package.json apparently we're using `date-fns`, `datejs` and `react-dates` (includes `moment`)

Reply:
I agree on the choice of avoiding a date library. Users of `ui-react` are then not bound by the choice of date library here + we don't have any real need for a date library. The bit I dislike is the magic of the regex.

Can we either:

- find a minimal date formatting library
- move the regex function into our own 'library' and add a few tests around the formatting